src/rsa/mod.rs

```diff
@@ -185,14 +179,15 @@ impl ReadFrom for PublicKey {
 
         let n = rsa::BigUint::from_bytes_be(n_buf);
         let e = rsa::BigUint::from_bytes_be(e_buf);
-        let public_key = RsaPublicKey::new(n, e)?;
+        let public_key = RSAPublicKey::new(n, e)?;
         Ok(PublicKey(public_key))
     }
 }
 
 impl Hash for PublicKey {
     fn hash<H: Hasher>(&self, state: &mut H) {
-        self.0.hash(state)
+        state.write(&self.0.n().to_bytes_be());
+        state.write(&self.0.e().to_bytes_be());
```

Comment:
Do we depend (or care) about stability of hashes across dependency upgrades? If not, I have no opinion on this and you can close the comment.

If we do care, then I _think_ this is likely to match the hash output if we ever upgrade to a newer RSA crate down the line:

```suggestion
        self.0.n().hash(state);
        self.0.e().hash(state);
```